discourse_rock/patches/sigterm.patch

```diff
@@ -15,14 +15,14 @@
 +
    unless initialized
      Discourse.preload_rails!
- 
-@@ -266,6 +275,9 @@ before_fork do |server, worker|
+
+@@ -299,6 +308,9 @@ before_fork do |server, worker|
  end
- 
+
  after_fork do |server, worker|
 +  Signal.trap 'TERM' do
 +    puts 'Unicorn worker intercepting TERM and doing nothing. Wait for master to sent QUIT'
 +  end
    DiscourseEvent.trigger(:web_fork_started)
    Discourse.after_fork
- end
+   SignalTrapLogger.instance.after_fork
```

Comment:
I'm seeing you put this here as v3.3.0 added a SignalTrapLogger and it is used because now they are trapping `USR2`. Perhaps it's better to place the patch nearer the other trap, maybe right bellow it?

v3.2.0
```ruby
after_fork do |server, worker|
  DiscourseEvent.trigger(:web_fork_started)
  Discourse.after_fork
end
```

v3.3.0
```ruby
after_fork do |server, worker|
  DiscourseEvent.trigger(:web_fork_started)
  Discourse.after_fork
  SignalTrapLogger.instance.after_fork

  Signal.trap("USR2") do
    message = <<~MSG
    Unicorn worker received USR2 signal indicating it is about to timeout, dumping backtrace for main thread
    #{Thread.current.backtrace&.join("\n")}
    MSG

    SignalTrapLogger.instance.log(Rails.logger, message, level: :warn)
  end
end
```

wdyt?

Reply:
Are you proposing this change to keep the code prettier? By putting traps back to back. Or does it serve a functional purpose?

So are you saying its better to do it like this?

```
after_fork do |server, worker|
  DiscourseEvent.trigger(:web_fork_started)
  Discourse.after_fork
  SignalTrapLogger.instance.after_fork

  Signal.trap("USR2") do
    message = <<~MSG
    Unicorn worker received USR2 signal indicating it is about to timeout, dumping backtrace for main thread
    #{Thread.current.backtrace&.join("\n")}
    MSG

    SignalTrapLogger.instance.log(Rails.logger, message, level: :warn)
  end
  Signal.trap 'TERM' do
    puts 'Unicorn worker intercepting TERM and doing nothing. Wait for master to sent QUIT'
  end
end
```

Reply:
It's purely cosmetical and up to you.